packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -102,28 +103,33 @@ export class QuoteRequestor {
 
         // create an array of promises for quote responses, using "undefined"
         // as a placeholder for failed requests.
+        const timeBeforeAwait = Date.now();
         const responsesIfDefined: Array<undefined | AxiosResponse<SignedOrder>> = await Promise.all(
-            this._rfqtMakerEndpoints.map(async rfqtMakerEndpoint => {
-                try {
-                    return await Axios.get<SignedOrder>(`${rfqtMakerEndpoint}/quote`, {
-                        headers: { '0x-api-key': _opts.apiKey },
-                        params: {
-                            takerAddress: _opts.takerAddress,
-                            ...inferQueryParams(marketOperation, makerAssetData, takerAssetData, assetFillAmount),
-                        },
-                        timeout: _opts.makerEndpointMaxResponseTimeMs,
-                    });
-                } catch (err) {
-                    this._warningLogger(
-                        `Failed to get RFQ-T firm quote from market maker endpoint ${rfqtMakerEndpoint} for API key ${
-                            _opts.apiKey
-                        } for taker address ${_opts.takerAddress}`,
-                    );
-                    this._warningLogger(err);
-                    return undefined;
+            Object.keys(this._rfqtAssetOfferings).map(async url => {
+                if (this._makerSupportsPair(url, makerAssetData, takerAssetData)) {
+                    try {
+                        return await Axios.get<SignedOrder>(`${url}/quote`, {
+                            headers: { '0x-api-key': _opts.apiKey },
+                            params: {
+                                takerAddress: _opts.takerAddress,
+                                ...inferQueryParams(marketOperation, makerAssetData, takerAssetData, assetFillAmount),
+                            },
+                            timeout: _opts.makerEndpointMaxResponseTimeMs,
+                        });
+                    } catch (err) {
+                        this._warningLogger(
+                            `Failed to get RFQ-T firm quote from market maker endpoint ${url} for API key ${
+                                _opts.apiKey
+                            } for taker address ${_opts.takerAddress}`,
+                        );
+                        this._warningLogger(err);
+                        return undefined;
+                    }
                 }
+                return undefined;
             }),
         );
+        this._infoLogger({ aggregatedRfqtLatencyMs: Date.now() - timeBeforeAwait });
```

Comment:
In a world where there are multiple RFQT providers: if this number is large -- how do we know which MM is responsible for it being large? I think ideally we would log the latency of each individual endpoint.

Reply:
When we first discussed logging this latency (I believe it was @PirosB3 who suggested it), I'm pretty sure we talked about doing it in aggregate like this.

My understanding is that gathering latency on individual responses isn't possible, because we're awaiting all of the responses in parallel.  If we were to position the code to be able to capture individual response times, I believe that we would be sacrificing the parallelism.

Reply:
It is possible without sacrificing the parallelism -- I will slack you about this

Reply:
Addressed in https://github.com/0xProject/0x-monorepo/pull/2574/commits/e42701599ad9d50d3f5ff258c826fcbbda9b1b26